sw/device/lib/dif/dif_sram_ctrl.h

```diff
@@ -11,256 +11,275 @@
+   * The "disabled" state.
    */
-  mmio_region_t base_addr;
-} dif_sram_ctrl_params_t;
-
-/**
- * A handle to SRAM Controller.
- *
- * This type should be treated as opaque by users.
- */
-typedef struct dif_sram_ctrl {
-  dif_sram_ctrl_params_t params;
-} dif_sram_ctrl_t;
+  kDifSramCtrlToggleDisabled,
+} dif_sram_ctrl_toggle_t;
 
 /**
- * The result of a SRAM Controller operation.
+ * SRAM Controller status.
  */
-typedef enum dif_sram_ctrl_result {
+typedef enum dif_sram_ctrl_status {
   /**
-   * Indicates that the operation succeeded.
+   * Bus integrity fault is detected. This error triggers a fatal_error alert.
+   * This condition is terminal.
    */
-  kDifSramCtrlOk = 0,
+  kDifSramCtrlStatusBusIntegErr = 0,
   /**
-   * Indicates some unspecified failure.
+   * Initialization counter has reached an invalid state. This error triggers
+   * a fatal_error alert. This condition is terminal.
    */
-  kDifSramCtrlError,
+  kDifSramCtrlStatusInitErr,
   /**
-   * Indicates that some parameter passed into a function failed a
-   * precondition.
-   *
-   * When this value is returned, no hardware operations occurred.
+   * SRAM Controller has received an escalate request, the scrambling keys have
+   * been reset to the default values and all subsequent memory requests will
+   * be blocked. This condition is terminal.
    */
-  kDifSramCtrlBadArg,
-} dif_sram_ctrl_result_t;
+  kDifSramCtrlStatusEscalated,
+  /**
+   * New scrambling key has been successfully obtained from OTP. If the flag is
+   * not set, the SRAM contents are still scrambled, but a default all-zero key
+   * and nonce are used to do so.
+   */
+  kDifSramCtrlStatusScrKeyValid,
+  /**
+   * Scrambling key has been derived from a valid key seed in OTP. When
+   * `kDifSramCtrlStatusScrKeyValid` is set, but this flag is unset - the
+   * scrambling key is still ephemeral (i.e., it is derived using entropy from
+   * CSRNG), but a default all-zero value is used as the key seed. This could
+   * happen when the scrambling key seeds have not yet been provisioned to OTP.
+   */
+  kDifSramCtrlStatusScrKeySeedValid,
+  /**
+   * Hardware initialization triggered via `dif_sram_ctrl_scramble` or
+   * `dif_sram_ctrl_wipe` has completed.
+   */
+  kDifSramCtrlStatusInitDone,
+} dif_sram_ctrl_status_t;
 
 /**
- * A toggle state: enabled, or disabled.
- *
- * This enum may be used instead of a `bool` when describing an enabled/disabled
- * state.
+ * SRAM Controller lockable functionality.
  */
-typedef enum dif_sram_ctrl_toggle {
+typedef enum dif_sram_ctrl_lock {
   /**
-   * The "enabled" state.
+   * SRAM scrambling key renewal and "wiping" lock, which includes the following
+   * API: `dif_sram_ctrl_scramble`, `dif_sram_ctrl_request_new_key`
+   * and `dif_sram_ctrl_wipe`.
    */
-  kDifSramCtrlToggleEnabled,
+  kDifSramCtrlLockCtrl = 0,
   /**
-   * The "disabled" state.
+   * Code execution from SRAM lock, which includes the following API:
```

Comment:
It is worth documenting that this setting may not be available depending on the OTP configuration of the chip (EN_SRAM_IFETCH fuse).

Reply:
Done.